storybook/ignore-console-messages.js

```diff
@@ -0,0 +1,168 @@
+			patterns: [ /.+/, /^$/ ],
+		},
+
+		// Components/MediaErrorHandler (all variants)
+		{
+			id: /^components-mediaerrorhandler--.+$/,
+			patterns: [ /.+/, /^$/ ],
+		},
+
+		// Components/WidgetErrorHandler › Default
+		{
+			id: /^components-widgeterrorhandler--default$/,
+			patterns: [ /.+/, /^$/ ],
+		},
+
+		// Modules/Analytics4/Components/AudienceSegmentation/Dashboard error stories
+		// - AudienceSelectionPanel › Audience sync error
+		// - AudienceSelectionPanel › Insufficient permissions error
+		// - SecondaryUserSetupWidget › Insufficient permission error
+		// - SecondaryUserSetupWidget › Setup error
+		{
+			id: /^modules-analytics4-components-audiencesegmentation-dashboard-(audienceselectionpanel--audience-sync-error|audienceselectionpanel--with-insufficient-permissions-error|secondaryusersetupwidget--insufficient-permission-error|secondaryusersetupwidget--setup-error)$/,
+			patterns: [
+				/^Google Site Kit API Error method:POST datapoint:sync-audiences/,
+			],
+		},
+	],
+};
+
+/**
+ * Normalizes a Puppeteer console/page message for robust matching.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} message Raw message text to normalize.
+ * @return {string} Normalized message text.
+ */
+export function normalizeMessage( message ) {
+	return String( message )
+		.replace( /JSHandle@(?:error|object)\s*/g, '' )
+		.trim();
+}
+
+/**
+ * Checks whether the given text matches any of the provided patterns.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string}   text     Text to test.
+ * @param {RegExp[]} patterns Array of regular expressions to test against.
+ * @return {boolean} True if any pattern matches; otherwise false.
+ */
+export function matchesAny( text, patterns ) {
+	return patterns.some( ( rx ) => rx.test( text ) );
+}
+
+/**
+ * Determines whether the given message should be ignored globally or for a given story.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} rawMessage Raw message text.
+ * @param {string} storyID    Storybook story ID (e.g. "components-errorhandler--default").
+ * @return {boolean} True if the message should be ignored; otherwise false.
+ */
+export function isIgnored( rawMessage, storyID ) {
+	const text = normalizeMessage( rawMessage );
+
+	// Global errors to ignore.
+	if ( matchesAny( text, IGNORE_CONSOLE_MESSAGES.global ) ) {
+		return true;
+	}
+
+	// Story-scoped expected errors.
+	if ( storyID ) {
+		for ( const entry of IGNORE_CONSOLE_MESSAGES.byStory ) {
+			if (
+				entry.id.test( storyID ) &&
+				matchesAny( text, entry.patterns )
+			) {
+				return true;
+			}
+		}
+	}
+
+	return false;
+}
+
+/**
+ * Extracts story ID from the current Storybook URL (?id=<storyId>).
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} url Full storybook URL.
+ * @return {string|null} Story ID if found; otherwise null.
+ */
+export function getStoryIDFromURL( url ) {
+	try {
+		const u = new URL( url );
+		return u.searchParams.get( 'id' );
+	} catch ( e ) {
+		return null;
+	}
```

Comment:
```suggestion
 * @param {string} urlString Full storybook URL.
 * @return {string|null} Story ID if found; otherwise null.
 */
export function getStoryIDFromURL( urlString ) {
	try {
		const url = new URL( urlString );
		return url.searchParams.get( 'id' );
	} catch ( error ) {
		return null;
	}
```